.gitlab-ci.yml

```diff
@@ -234,13 +259,35 @@ test:prep:
     - mender-qa-worker-generic-light
 
 build:docker:
-  after_script:
-    - docker build --tag ${DOCKER_REPOSITORY}:base --target base --build-arg GIT_COMMIT_TAG="${DOCKER_PUBLISH_COMMIT_TAG}" .
-    - docker save ${DOCKER_REPOSITORY}:base > baseImage.tar
+  services:
+    - docker:dind
+  # unfortunately we can't just extend the after_script here as it would require rebuilding all layers of the image for the different build targets
+  script:
+    - echo "building ${CI_PROJECT_NAME} for ${DOCKER_BUILD_SERVICE_IMAGE}"
+    - apk add --no-cache git
+    - docker context create the-context
+    - docker buildx create --driver=docker-container --name=buildx-builder --use the-context
+    - docker buildx build
+      --build-arg GIT_COMMIT_TAG="${DOCKER_PUBLISH_COMMIT_TAG}"
+      --cache-from=type=local,src=dockerCache
+      --cache-to=type=local,dest=dockerCache
+      --tag $DOCKER_BUILD_SERVICE_IMAGE
+      --output=type=docker,dest=- . > image.tar
```

Comment:
@oldgiova I went slightly overboard with this as I was trying to prevent layer rebuilds in the other images I build right afterwards, but the `buildx` setup might be helpful for the multi-arch route you wanted to look into 👍

the layer rebuilds turned out to be due to the resulting tar files + the cache I set up 🙈 => thus the changes in the `.dockerignore`